AutoRest/Generators/CSharp/CSharp.Tests/Expected/AcceptanceTests/CustomBaseUriMoreOptions/Paths.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+// 
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+namespace Fixtures.AcceptanceTestsCustomBaseUriMoreOptions
+{
+    using System;
+    using System.Linq;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Rest;
+    using Microsoft.Rest.Serialization;
+    using Newtonsoft.Json;
+    using Models;
+
+    /// <summary>
+    /// Paths operations.
+    /// </summary>
+    public partial class Paths : IServiceOperations<AutoRestParameterizedCustomHostTestClient>, IPaths
+    {
+        /// <summary>
+        /// Initializes a new instance of the Paths class.
+        /// </summary>
+        /// <param name='client'>
+        /// Reference to the service client.
+        /// </param>
+        public Paths(AutoRestParameterizedCustomHostTestClient client)
+        {
+            if (client == null) 
+            {
+                throw new ArgumentNullException("client");
+            }
+            this.Client = client;
+        }
+
+        /// <summary>
+        /// Gets a reference to the AutoRestParameterizedCustomHostTestClient
+        /// </summary>
+        public AutoRestParameterizedCustomHostTestClient Client { get; private set; }
+
+        /// <summary>
+        /// Get a 200 to test a valid base uri
+        /// </summary>
+        /// <param name='vault'>
+        /// The vault name, e.g. https://myvault
+        /// </param>
+        /// <param name='secret'>
+        /// Secret value.
+        /// </param>
+        /// <param name='keyName'>
+        /// The key name with value 'key1'.
+        /// </param>
+        /// <param name='keyVersion'>
+        /// The key version. Default value 'v1'.
+        /// </param>
+        /// <param name='customHeaders'>
+        /// Headers that will be added to request.
+        /// </param>
+        /// <param name='cancellationToken'>
+        /// The cancellation token.
+        /// </param>
+        /// <return>
+        /// A response object containing the response body and response headers.
+        /// </return>
+        public async Task<HttpOperationResponse> GetEmptyWithHttpMessagesAsync(string vault, string secret, string keyName, string keyVersion = "v1", Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
```

Comment:
OK, good, I now understand why the feature was designed this way. I definitely would make 'first' a default.